core/contracts/oracle/implementation/Governor.sol

```diff
@@ -60,10 +60,20 @@ contract Governor is MultiRole, Testable {
      * @param _timerAddress Contract that stores the current time in a testing environment.
      * Must be set to 0x0 for production environments that use live time.
      */
-    constructor(address _finderAddress, address _timerAddress) public Testable(_timerAddress) {
+    constructor(address _finderAddress, uint startingId, address _timerAddress) public Testable(_timerAddress) {
```

Comment:
Do we not perhaps want to store this `startingId` as a state variable within the `Governor`? Say for arguments sake we've got 15 proposals currently. We use the method `getProposal(n)` to get the `n`th proposal. If the starting proposal is in position 0 this is easy to do as it's index is simply `15-1=14` as we are indexed from 0. 

However if you shift the starting point of the index it's not not immediately clear where this index should be. For example say had constructed the Governor with a `startingId` of 200 the 15th position is 214 which is strange to reason about unless you know the starting point of 200.

I might be missing something here around how we can read from this state but it seems a bit counter intuitive.

Reply:
I'm a little confuses about what you're suggesting... who would need to compute and subtract the offset? The voters who want to look up proposals? Let me try to offer an alternative explanation.

I think the `getProposals(n)` is supposed to get the proposal by its *id*. So if the previous governor's last id is 5, we want to start the next governor's ids at 6. Since the array is set to 6, when the first proposal is made, it will have an id of 6, and voters will retrieve it by calling `getProposals(6)`.

The idea is to avoid any offset math (by the users or the governor) by just leaving empty slots in the array. To help voters, though, we want the ids to pick up where they left off after upgrading the governor, so `Admin 1` is a unique proposal. The id the voter should use for lookup will show up as the identifier for the price request, so they won't have to perform any offset math to query it. The id will also be emitted as part of an event during the proposal so the proposer can get it.

Reply:
Btw, these are good questions, and it probably means we need better documentation around this.

Reply:
Ok cool. Being able to retrieve this from events makes this better. My main concern was around knowing how large the offset was for a given governor instance such that you know where to start looking from for the next proposal _without_ needing to traverse the array over the blank spots.